src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+package mate.academy.bookstore.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.order.CreateOrderRequestDto;
+import mate.academy.bookstore.dto.order.OrderResponseDto;
+import mate.academy.bookstore.dto.order.UpdateOrderDto;
+import mate.academy.bookstore.dto.orderitem.OrderItemDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.CartItemMapper;
+import mate.academy.bookstore.mapper.OrderItemMapper;
+import mate.academy.bookstore.mapper.OrderMapper;
+import mate.academy.bookstore.mapper.ShoppingCartMapper;
+import mate.academy.bookstore.model.CartItem;
+import mate.academy.bookstore.model.Order;
+import mate.academy.bookstore.model.OrderItem;
+import mate.academy.bookstore.model.ShoppingCart;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.order.OrderRepository;
+import mate.academy.bookstore.repository.orderitem.OrderItemRepository;
+import mate.academy.bookstore.repository.shopingcart.ShoppingCartRepository;
+import mate.academy.bookstore.service.OrderService;
+import mate.academy.bookstore.service.ShoppingCartService;
+import mate.academy.bookstore.service.UserService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserService userService;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final Order order = new Order();
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    @Transactional
+    public void createUserOrder(Long userId, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartService.getShoppingCartModel(userId);
```

Comment:
You can simplify this logic
1. Find shoppingCart
```
2. if (cart.getCartItems().isEmpty()) {
            throw new OrderProcessingException("Cart is empty for user: " + userId);
}
```
3. Order order = orderMapper.cartToOrder(cart, orderDto.shippingAddress());
4. shoppingCart.getCartItems().clear();
5. save

Where in OrderMapper
```
@Mapping(target = "id", ignore = true)
@Mapping(target = "total", source = "cart.cartItems", qualifiedByName = "total")
@Mapping(target = "orderItems", source = "cart.cartItems")
Order cartToOrder(ShoppingCart cart, String shippingAddress);
```

and 
```
@Named("total")
    default BigDecimal getTotal(Set<CartItem> cartItems) {
        return cartItems.stream()
                .map(i -> i.getBook().getPrice().multiply(BigDecimal.valueOf(i.getQuantity())))
                .reduce(BigDecimal.ZERO, BigDecimal::add);
    }
```

Also, you can initialise `Status = Status.PENDING and orderDate = LocalDateTime.now()` in Order entity